src/fastlane/release_actions/lib/fastlane/plugin/release_actions/helper/version/acceptor.rb

```diff
@@ -0,0 +1,167 @@
+class Version
+  # Symbol holds details about the alphabet recognized by the Acceptor. It translates
+  # characters into the an enumeration value for use in determining the right transition.
+  module Symbol
+    POSITIVE_INTEGER = 0
+    ZERO = 1
+    LETTER = 2
+    DOT = 3
+    DASH = 4
+    PLUS = 5
+    UNKNOWN = 99
+
+    def self.for(character)
+      case character
+      when ('1'..'9')
+        Symbol::POSITIVE_INTEGER
+      when '0'
+        Symbol::ZERO
+      when ('A'..'Z'), ('a'..'z')
+        Symbol::LETTER
+      when '.'
+        Symbol::DOT
+      when '-'
+        Symbol::DASH
+      when '+'
+        Symbol::PLUS
+      else
+        Symbol::UNKNOWN
+      end
+    end
+  end
+
+  # Acceptor is a deterministic finite automaton (DFA) that validates strings as
+  # semantic versions. A DFA is a graph where the vertices are states (e.g. major,
+  # minor, etc.) and the edges are transitions (e.g. start -> major). The DFA moves
+  # between states using the transitions defined in the STATES constant as it processes
+  # the characters of the input string. Certain states are designated as accepting,
+  # meaning the state is a valid endpoint for the grammar being parsed. If the DFA
+  # terminates in an accepting state, the string is valid and it accepts the
+  # input. If the DFA cannot make a transition, finds an invalid character, or
+  # terminates in a non-accepting state, the string is not valid. A simplified diagram
+  # of the state machine can be found in this repo in
+  # `docs/version-acceptor-dfa.{png,svg}` (drawn using http://madebyevan.com/fsm/).
+  # While there's some complexity in maintaining the table of edges, it provides O(n)
+  # runtime performance and has a maintainability edge over regular expressions.
+  #
+  # Acceptor is compliant with Semantic Versioning 2.0.0. For more details on the
+  # specification, see: https://semver.org/.
+  #
+  # ==== Examples
+  #
+  #   Version::Acceptor.new('1.0.0').valid?            # => true
+  #   Version::Acceptor.new('1.0.0-unstable.0').valid? # => true
+  #   Version::Acceptor.new('1.0-alpha+234').valid?    # => true
+  #   Version::Acceptor.new('1.0318.2').valid?         # => false
+  #   Version::Acceptor.new('1.0.PRO).valid?           # => false
+  #   Version::Acceptor.new('tetris').valid?           # => false
+  class Acceptor
+    ACCEPTING = [:patch, :patch_end, :prerelease, :build]
+
+    START = :start
+
+    STATES = {
+      start: {
+        Symbol::POSITIVE_INTEGER => :major,
+        Symbol::ZERO => :major_end
```

Comment:
I assume that a version like `0.1.0` would correctly parse the major version as `0` as opposed to `nil`?

Reply:
Correct. Should be a test enforcing this, I believe.